extensions/ql-vscode/src/view/results/alert-table.tsx

```diff
@@ -302,34 +306,103 @@ export class PathTable extends React.Component<PathTableProps, PathTableState> {
     </table>;
   }
 
-  private handleNavigationEvent(event: NavigationEvent) {
+  private handleNavigationEvent(event: NavigateMsg) {
     this.setState(prevState => {
-      const { selectedPathNode } = prevState;
-      if (selectedPathNode === undefined) return prevState;
-
-      const path = Keys.getPath(this.props.resultSet.interpretation.data, selectedPathNode);
-      if (path === undefined) return prevState;
-
-      const nextIndex = selectedPathNode.pathNodeIndex + event.direction;
-      if (nextIndex < 0 || nextIndex >= path.locations.length) return prevState;
+      const key = this.getNewSelection(prevState.selectedItem, event.direction);
+      const data = this.props.resultSet.interpretation.data;
+
+      // Check if the selected node actually exists (bounds check) and get its location if relevant
+      let jumpLocation: Sarif.Location | undefined;
+      if (key.pathNodeIndex != null) {
+        jumpLocation = Keys.getPathNode(data, key);
+        if (jumpLocation == null) {
+          return prevState; // Result does not exist
+        }
+      } else if (key.pathIndex != null) {
+        if (Keys.getPath(data, key) == null) {
+          return prevState; // Path does not exist
+        }
+        jumpLocation = undefined; // When selecting a 'path', don't jump anywhere.
+      } else {
+        jumpLocation = Keys.getResult(data, key)?.locations?.[0];
+        if (jumpLocation == null) {
+          return prevState; // Path step does not exist.
+        }
+      }
+      if (jumpLocation !== undefined) {
```

Comment:
Is there a semantic difference between null and undefined here? I see you are using `== null` and `!= null` above, but `!== undefined` here. I personally prefer just using `if (jumpLocation) ...` or `if (!jumpLocation) ...`, which I think is clearer than using obscure JS semantics that `undefined == null`. 

This is just a preference of mine, so it's not a requirement to move forward with this PR.

Reply:
I've updated all the checks to consistently check for `undefined` only.

This seems to be the convention here, as there are >200 occurrences of `=== undefined` or `!== undefined` and only 16 of `== null` and `!= null`.